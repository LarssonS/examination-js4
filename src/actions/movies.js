import database from '../firebase/firebase';

//ADD_MOVIE

export const addMovie = (movie) => ({
    type: 'ADD_MOVIE',
    movie 
});

export const startAddMovie = (movieData = {}) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        const {
            title = '',
            description = '',
            director = '',
            genre = '',
            filmWrapper = '',
            raiting = 0,
            createdAt = 0
        } = movieData;
        const movie = { title, description, director, genre, filmWrapper, raiting, createdAt, uid };

        return database.ref(`users/movies`).push(movie).then(() => {
            dispatch(addMovie({ ...movie }));
          });
    };
};

export const removeMovie = ({ id } = {}) => ({
    type: 'REMOVE_MOVIE',
    id
});

export const startRemoveMovie = ({ id } = {}) => {
    return(dispatch) => {
        return database.ref(`users/movies/${id}`).remove().then(() => {
            dispatch(removeMovie({ id }));
        });
    }
};

export const editMovie = ({id, updates}) => ({
    type: 'EDIT_MOVIE',
    id,
    updates
});

export const startEditMovie = ({id, updates}) => {
    return(dispatch) => {
        return database.ref(`users/movies/${id}`).update(updates).then(() => {
           dispatch(editMovie({id, updates})); 
        });
    };
};

export function fetchListenerMovies(ref, callback) {
    ref.on("child_added", (snapshot) => {
            const data = { ...snapshot.val(), id: snapshot.key };
            callback(data);
        });
};

export function removeListenerMovie(ref, callback) {
    ref.on("child_removed", (snapshot) => {
            callback(snapshot.key);
        });
};

export function changedListenerMovie(ref, callback) {
    ref.on("child_changed", (snapshot) => {
            const data = { updates: snapshot.val(), id: snapshot.key };
            callback(data);
        });
};
