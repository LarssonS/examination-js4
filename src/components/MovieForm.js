import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import { Container, Form, FormGroup, Label, Input, Button } from 'reactstrap';

export default class MovieForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            title: props.movie ? props.movie.title : '',
            description: props.movie ? props.movie.description : '',
            director: props.movie ? props.movie.director : '',
            genre: props.movie ? props.movie.genre : '',
            filmWrapper: props.movie ? props.movie.filmWrapper : '',
            raiting: props.movie ? props.movie.raiting : 0,
            createdAt: props.movie ? moment(props.movie.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    }

    onTitleChange = (e) => {
        const title = e.target.value;
        this.setState(() => ({ title }));
    };
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onDirectorChange = (e) => {
        const director = e.target.value;
        this.setState(() => ({ director }));
    };
    onGenreChange = (e) => {
        const genre = e.target.value;
        this.setState(() => ({ genre }));
    };
    onFilmWrapperChange = (e) => {
        const filmWrapper = e.target.value;
        this.setState(() => ({ filmWrapper }));
    };
    onRaitingChange = (e) => {
        const raiting = e.target.value;
        if(raiting > 0 && raiting <= 5) {
            this.setState(() => ({ raiting }));
        }
    };
    onDateChange = (createdAt) => {
        if(createdAt) {
            this.setState(() => ({ createdAt }));
        }
    };
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };
    onSubmit = (e) => {
        e.preventDefault();

        if(
            !this.state.title ||
            !this.state.description ||
            !this.state.director ||
            !this.state.genre ||
            !this.state.filmWrapper ||
            !this.state.raiting ||
            !this.state.createdAt
         ) {
            this.setState(() => ({ error: 'Please fill in the blanks' }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                title: this.state.title,
                description: this.state.description,
                director: this.state.director,
                genre: this.state.genre,
                filmWrapper: this.state.filmWrapper,
                raiting: this.state.raiting,
                createdAt: this.state.createdAt.valueOf()
            });
        }
    };

    render() {
        return (
            <div>
                <Form onSubmit={this.onSubmit}>
                    {this.state.error && <p>{this.state.error}</p>}
                    <FormGroup>
                        <Label>Movie title</Label>
                        <Input
                            type="text"
                            placeholder="Title"
                            autoFocus
                            value={this.state.title}
                            bsSize="lg"
                            onChange={this.onTitleChange}
                        />
                    </FormGroup>
                    <FormGroup>
                        <Label>Description</Label>
                        <Input
                            type="text"
                            placeholder="Description"
                            value={this.state.description}
                            bsSize="lg"
                            onChange={this.onDescriptionChange}
                        />
                    </FormGroup>
                    <FormGroup>
                        <Label>Director</Label>
                        <Input
                            type="text"
                            placeholder="Director"
                            value={this.state.director}
                            bsSize="lg"
                            onChange={this.onDirectorChange}
                        />
                    </FormGroup>
                    <FormGroup>
                        <Label>Movie genre</Label>
                        <Input type="select" bsSize="lg" onChange={this.onGenreChange}>
                            <option disabled>Select genre</option>
                            <option>None</option>
                            <option value="action">Action</option>
                            <option value="comedy">Comedy</option>
                            <option value="sci-fi">Sci-Fi</option>
                            <option value="war">War</option>
                            <option value="drama">Drama</option>
                            <option value="sport">Sport</option>
                            <option value="documentary">Documentary</option>
                        </Input>
                    </FormGroup>
                    <FormGroup>
                        <Label>Film image address</Label>
                        <Input
                            type="text"
                            placeholder="Film wrapper"
                            value={this.state.filmWrapper}
                            bsSize="lg"
                            onChange={this.onFilmWrapperChange}
                        />
                    </FormGroup>
                    <FormGroup>
                        <Label>Movie raiting</Label>
                        <Input
                            type="number" min="1" max="5"
                            placeholder="Rate"
                            value={this.state.raiting}
                            bsSize="lg"
                            onChange={this.onRaitingChange}
                        />
                    </FormGroup>
                    <FormGroup>
                        <SingleDatePicker
                            date={this.state.createdAt}
                            onDateChange={this.onDateChange}
                            focused={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                        />
                    </FormGroup>
                    <FormGroup>
                        {this.props.movie ? <Button size="lg">Update movie</Button> : <Button size="lg">Add Movie</Button>}
                    </FormGroup>
                </Form>
            </div>
        )
    }
}