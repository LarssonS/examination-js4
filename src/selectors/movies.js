import moment from 'moment';

//Get visible movies
export default (movies, { text, sortBy, startDate, endDate }) => {
    return movies.filter((movie) => {
        const createdAtMoment = moment(movie.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const titleMatch = movie.title.toLowerCase().includes(text.toLowerCase());
        const genreMatch = movie.genre.toLowerCase().includes(sortBy.toLowerCase());

        return startDateMatch && endDateMatch && titleMatch && genreMatch;
    });
}