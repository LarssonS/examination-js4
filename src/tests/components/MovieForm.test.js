import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import MovieForm from '../../components/MovieForm';
import movies from '../fixtures/movies';

test('should render MovieForm correctly', () => {
  const wrapper = shallow(<MovieForm />);
  expect(wrapper).toMatchSnapshot();
});

test('should render MovieForm correctly with movie data', () => {
  const wrapper = shallow(<MovieForm movie={movies[1]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render error for invalid form submission', () => {
  const wrapper = shallow(<MovieForm />);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('Form').simulate('submit', {
    preventDefault: () => { }
  });
  expect(wrapper.state('error').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test('should set title on input change', () => {
  const value = 'New title';
  const wrapper = shallow(<MovieForm />);
  wrapper.find('Input').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('title')).toBe(value);
});

test('should set description on input change', () => {
  const value = 'New description';
  const wrapper = shallow(<MovieForm />);
  wrapper.find('Input').at(1).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('description')).toBe(value);
});

test('should set director on input change', () => {
  const value = 'New director';
  const wrapper = shallow(<MovieForm />);
  wrapper.find('Input').at(2).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('director')).toBe(value);
});

test('should call onSubmit prop for valid form submission', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<MovieForm movie={movies[0]} onSubmit={onSubmitSpy} />);
  wrapper.find('Form').simulate('submit', {
    preventDefault: () => { }
  });
  expect(wrapper.state('error')).toBe('');
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
    title: movies[0].title,
    description: movies[0].description,
    director: movies[0].director,
    genre: movies[0].genre,
    filmWrapper: movies[0].filmWrapper,
    raiting: movies[0].raiting,
    createdAt: movies[0].createdAt
  });
});

test('should set new date on date change', () => {
  const now = moment();
  const wrapper = shallow(<MovieForm />);
  wrapper.find('SingleDatePicker').prop('onDateChange')(now);
  expect(wrapper.state('createdAt')).toEqual(now);
});

test('should set calendar focus on change', () => {
  const focused = true;
  const wrapper = shallow(<MovieForm />);
  wrapper.find('SingleDatePicker').prop('onFocusChange')({ focused });
  expect(wrapper.state('calendarFocused')).toBe(focused);
});
