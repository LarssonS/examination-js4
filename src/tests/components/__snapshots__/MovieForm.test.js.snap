// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieForm correctly 1`] = `
<div>
  <Form
    onSubmit={[Function]}
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie title
      </Label>
      <Input
        autoFocus={true}
        bsSize="lg"
        onChange={[Function]}
        placeholder="Title"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Description
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Description"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Director
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Director"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie genre
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        type="select"
      >
        <option
          disabled={true}
        >
          Select genre
        </option>
        <option>
          None
        </option>
        <option
          value="action"
        >
          Action
        </option>
        <option
          value="comedy"
        >
          Comedy
        </option>
        <option
          value="sci-fi"
        >
          Sci-Fi
        </option>
        <option
          value="war"
        >
          War
        </option>
        <option
          value="drama"
        >
          Drama
        </option>
        <option
          value="sport"
        >
          Sport
        </option>
        <option
          value="documentary"
        >
          Documentary
        </option>
      </Input>
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Film image address
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Film wrapper"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie raiting
      </Label>
      <Input
        bsSize="lg"
        max="5"
        min="1"
        onChange={[Function]}
        placeholder="Rate"
        type="number"
        value={0}
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <SingleDatePicker
        anchorDirection="left"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1970-01-01T00:00:00.000Z"}
        daySize={39}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        id="date"
        initialVisibleMonth={null}
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        orientation="horizontal"
        phrases={
          Object {
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month",
            "jumpToPrevMonth": "Move backward to switch to the previous month",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
            "moveFocusByOneMonth": "Switch months",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
            "openThisPanel": "Open this panel",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field",
            "selectFocusedDate": "Select the date in focus",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        renderCalendarInfo={null}
        renderDay={null}
        renderMonth={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        withFullScreenPortal={false}
        withPortal={false}
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Button
        color="secondary"
        size="lg"
        tag="button"
      >
        Add Movie
      </Button>
    </FormGroup>
  </Form>
</div>
`;

exports[`should render MovieForm correctly with movie data 1`] = `
<div>
  <Form
    onSubmit={[Function]}
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie title
      </Label>
      <Input
        autoFocus={true}
        bsSize="lg"
        onChange={[Function]}
        placeholder="Title"
        type="text"
        value="spiderman"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Description
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Description"
        type="text"
        value="spider"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Director
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Director"
        type="text"
        value="secret"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie genre
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        type="select"
      >
        <option
          disabled={true}
        >
          Select genre
        </option>
        <option>
          None
        </option>
        <option
          value="action"
        >
          Action
        </option>
        <option
          value="comedy"
        >
          Comedy
        </option>
        <option
          value="sci-fi"
        >
          Sci-Fi
        </option>
        <option
          value="war"
        >
          War
        </option>
        <option
          value="drama"
        >
          Drama
        </option>
        <option
          value="sport"
        >
          Sport
        </option>
        <option
          value="documentary"
        >
          Documentary
        </option>
      </Input>
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Film image address
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Film wrapper"
        type="text"
        value="img"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie raiting
      </Label>
      <Input
        bsSize="lg"
        max="5"
        min="1"
        onChange={[Function]}
        placeholder="Rate"
        type="number"
        value={7}
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <SingleDatePicker
        anchorDirection="left"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1969-12-28T00:00:00.000Z"}
        daySize={39}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        id="date"
        initialVisibleMonth={null}
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        orientation="horizontal"
        phrases={
          Object {
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month",
            "jumpToPrevMonth": "Move backward to switch to the previous month",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
            "moveFocusByOneMonth": "Switch months",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
            "openThisPanel": "Open this panel",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field",
            "selectFocusedDate": "Select the date in focus",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        renderCalendarInfo={null}
        renderDay={null}
        renderMonth={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        withFullScreenPortal={false}
        withPortal={false}
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Button
        color="secondary"
        size="lg"
        tag="button"
      >
        Update movie
      </Button>
    </FormGroup>
  </Form>
</div>
`;

exports[`should render error for invalid form submission 1`] = `
<div>
  <Form
    onSubmit={[Function]}
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie title
      </Label>
      <Input
        autoFocus={true}
        bsSize="lg"
        onChange={[Function]}
        placeholder="Title"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Description
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Description"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Director
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Director"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie genre
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        type="select"
      >
        <option
          disabled={true}
        >
          Select genre
        </option>
        <option>
          None
        </option>
        <option
          value="action"
        >
          Action
        </option>
        <option
          value="comedy"
        >
          Comedy
        </option>
        <option
          value="sci-fi"
        >
          Sci-Fi
        </option>
        <option
          value="war"
        >
          War
        </option>
        <option
          value="drama"
        >
          Drama
        </option>
        <option
          value="sport"
        >
          Sport
        </option>
        <option
          value="documentary"
        >
          Documentary
        </option>
      </Input>
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Film image address
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Film wrapper"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie raiting
      </Label>
      <Input
        bsSize="lg"
        max="5"
        min="1"
        onChange={[Function]}
        placeholder="Rate"
        type="number"
        value={0}
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <SingleDatePicker
        anchorDirection="left"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1970-01-01T00:00:00.000Z"}
        daySize={39}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        id="date"
        initialVisibleMonth={null}
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        orientation="horizontal"
        phrases={
          Object {
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month",
            "jumpToPrevMonth": "Move backward to switch to the previous month",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
            "moveFocusByOneMonth": "Switch months",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
            "openThisPanel": "Open this panel",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field",
            "selectFocusedDate": "Select the date in focus",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        renderCalendarInfo={null}
        renderDay={null}
        renderMonth={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        withFullScreenPortal={false}
        withPortal={false}
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Button
        color="secondary"
        size="lg"
        tag="button"
      >
        Add Movie
      </Button>
    </FormGroup>
  </Form>
</div>
`;

exports[`should render error for invalid form submission 2`] = `
<div>
  <Form
    onSubmit={[Function]}
    tag="form"
  >
    <p>
      Please fill in the blanks
    </p>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie title
      </Label>
      <Input
        autoFocus={true}
        bsSize="lg"
        onChange={[Function]}
        placeholder="Title"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Description
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Description"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Director
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Director"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie genre
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        type="select"
      >
        <option
          disabled={true}
        >
          Select genre
        </option>
        <option>
          None
        </option>
        <option
          value="action"
        >
          Action
        </option>
        <option
          value="comedy"
        >
          Comedy
        </option>
        <option
          value="sci-fi"
        >
          Sci-Fi
        </option>
        <option
          value="war"
        >
          War
        </option>
        <option
          value="drama"
        >
          Drama
        </option>
        <option
          value="sport"
        >
          Sport
        </option>
        <option
          value="documentary"
        >
          Documentary
        </option>
      </Input>
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Film image address
      </Label>
      <Input
        bsSize="lg"
        onChange={[Function]}
        placeholder="Film wrapper"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Movie raiting
      </Label>
      <Input
        bsSize="lg"
        max="5"
        min="1"
        onChange={[Function]}
        placeholder="Rate"
        type="number"
        value={0}
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <SingleDatePicker
        anchorDirection="left"
        customCloseIcon={null}
        customInputIcon={null}
        date={"1970-01-01T00:00:00.000Z"}
        daySize={39}
        disabled={false}
        displayFormat={[Function]}
        enableOutsideDays={false}
        firstDayOfWeek={null}
        focused={false}
        hideKeyboardShortcutsPanel={false}
        horizontalMargin={0}
        id="date"
        initialVisibleMonth={null}
        isDayBlocked={[Function]}
        isDayHighlighted={[Function]}
        isOutsideRange={[Function]}
        isRTL={false}
        keepOpenOnDateSelect={false}
        monthFormat="MMMM YYYY"
        navNext={null}
        navPrev={null}
        numberOfMonths={1}
        onClose={[Function]}
        onDateChange={[Function]}
        onFocusChange={[Function]}
        onNextMonthClick={[Function]}
        onPrevMonthClick={[Function]}
        orientation="horizontal"
        phrases={
          Object {
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month",
            "jumpToPrevMonth": "Move backward to switch to the previous month",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
            "moveFocusByOneMonth": "Switch months",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
            "openThisPanel": "Open this panel",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field",
            "selectFocusedDate": "Select the date in focus",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
            "upArrowDownArrow": "up and down arrow keys",
          }
        }
        placeholder="Date"
        readOnly={false}
        renderCalendarInfo={null}
        renderDay={null}
        renderMonth={null}
        reopenPickerOnClearDate={false}
        required={false}
        screenReaderInputMessage=""
        showClearDate={false}
        showDefaultInputIcon={false}
        withFullScreenPortal={false}
        withPortal={false}
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Button
        color="secondary"
        size="lg"
        tag="button"
      >
        Add Movie
      </Button>
    </FormGroup>
  </Form>
</div>
`;
